### PR TITLE
[MINOR] [VL] Enhance the gluten timer to support seconds, milliseconds, and microseconds

### PR DESCRIPTION
**Pull request by kecookier**
## What changes were proposed in this pull request?
Sometimes we need a timer in seconds, e.g., for measuring file write time. Enhance the gluten timer to support seconds, milliseconds, microseconds and nanoseconds.

## How was this patch tested?
Exist CI.

### CODE REVIEW

**Comment by kecookier** [comment]
@marin-ma Can you help review this?

File: cpp/core/shuffle/Payload.cc [hunk 1]
```diff
@@ -310,7 +310,7 @@ arrow::Result<std::vector<std::shared_ptr<arrow::Buffer>>> BlockPayload::deseria
   RETURN_NOT_OK(inputStream->Read(sizeof(uint32_t), &numRows));
   uint32_t numBuffers;
   RETURN_NOT_OK(inputStream->Read(sizeof(uint32_t), &numBuffers));
-  timer.reset();
+  deserializeTime += timer.realTimeUsed();
```
Comment (marin-ma) [inline cpp/core/shuffle/Payload.cc:313]: Why do we need this change? timer.start() and timer.stop() is not called. the realTimeUsed would be zero

Comment (kecookier) [reply cpp/core/shuffle/Payload.cc]: @marin-ma  Thanks, Um.. I've added this to fix the compile error with `std::unique_ptr<ScopedTimer>`. It was a typo. Now, I have fixed it in another way.

**Comment by kecookier** [comment]
@marin-ma The comments have been fixed. Could you please review them again?

**Review by FelixYBW — APPROVED** [review]

**Pull request merged**